lib/unique_elements.rb

```diff
@@ -1,6 +1,21 @@
 # This method takes an array of elements (these might be strings, integers, floats, or a
 # combination of the above), and returns an array of the elements that appear once and only
 # once.
-def find_unique_elements(arr)
+
+def count_elements(arr)
+  h={}
+  arr.each do |n|
+    if h.has_key?(n) == false
+      h[n]=1
+    elsif h.has_key?(n) == true
+      h[n]= h[n] +1
+    end
+  end
+  
   
+def find_unique_elements(arr)
+    a = count_elements(arr)
+    #next line should create an array that stores the hash keys with a value of '1' from the hash created by the 'count_elements' method
```

Comment:
Your thought process–the algorithmic thinking–is right on. But the code you've written doesn't achieve that. (This is okay! The coding part is much easier to learn than the algorithmic part, so you're in good shape.)

I suggest doing this. Open a fresh [repl.it](https://repl.it/languages/ruby) and paste this code into it:

```ruby
frequency_hash = {
  "banana" => 4,
  "apple" => 2,
  "fig" => 1,
  "strawberry" => 49,
  "date" => 1,
  "pear" => 3
}

array_with_unique_elements = []

# ...

puts array_with_unique_elements.inspect
```

(Don't worry about the `.inspect` part.)

The above code defines a hash with elements and their frequency. Just like your `count_elements` method does.

The last line displays an array that only contains the unique elements from that hash. So it should be an array with just `fig` and `date` in it. But it doesn't work yet! **You** need to figure out how to get that array actually populated.

Obviously you could cheat and do `array_with_unique_elements = ["fig", "date"]` – but try to implement it such that your code would work with a different set of values too.

Your code should go where the `# ...` line is.

---

Notice that this process is exactly the process your idea/algorithm explains. Hopefully doing this work in Repl.it will help you get some clarity about the code needed to make it work.